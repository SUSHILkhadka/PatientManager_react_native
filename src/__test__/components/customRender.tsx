import {NavigationContainer} from '@react-navigation/native';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {render} from '@testing-library/react-native';
import React, {PropsWithChildren} from 'react';
import {Provider} from 'react-redux';

import {allReducers, RootState} from '../../redux_toolkit/stores/store';

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: allReducers,
    preloadedState,
  });
};

export function renderWithProvidersAndNavigation(
  ui: React.ReactElement,
  {preloadedState = {}, store = setupStore(preloadedState), ...renderOptions}: any = {},
) {
  function Wrapper({children}: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <NavigationContainer>{children}</NavigationContainer>
      </Provider>
    );
  }
  return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
