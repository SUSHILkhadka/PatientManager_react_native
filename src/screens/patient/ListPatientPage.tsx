import {useNavigation} from '@react-navigation/native';
import React, {useLayoutEffect} from 'react';
import {SafeAreaView, StyleSheet, TouchableOpacity, View} from 'react-native';
import Ionicons from 'react-native-vector-icons/Ionicons';
import PatientTable from '../../components/patient/PatientTable';
import {COLOR} from '../../components/styles/constants';
import {typeOfUseNavigationHook} from '../../navigator/Navigator';
import {safeAreaStyles} from '../login/LoginPage';

const ListPatientPage = () => {
  const navigation: typeOfUseNavigationHook['navigation'] = useNavigation();

  useLayoutEffect(() => {
    navigation.setOptions?.({
      headerRight: () => {
        return (
          <TouchableOpacity onPress={() => navigation.navigate('add')}>
            <Ionicons style={styles.addIcon_text} name="person-add" />
          </TouchableOpacity>
        );
      },
    });
  });

  return (
    <SafeAreaView style={safeAreaStyles.page}>
      <View>
        <PatientTable />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  addIcon_text: {
    color: COLOR.pink1,
    fontSize: 32,
    alignSelf: 'center',
    marginRight: 16,
  },
});

export default ListPatientPage;
