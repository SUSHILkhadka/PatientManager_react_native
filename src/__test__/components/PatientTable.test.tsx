import React from 'react';
import {act} from 'react-test-renderer';
import ListPatientPage from '../../screens/patient/ListPatientPage';
import {renderWithProvidersAndNavigation} from './customRender';

jest.mock('../../axios/api');
it('should render list vaccine page properly', async () => {
  const page = renderWithProvidersAndNavigation(<ListPatientPage />);

  await act(async () => {
    page.rerender(<ListPatientPage />);
  });
  expect(page).toMatchSnapshot();
});
