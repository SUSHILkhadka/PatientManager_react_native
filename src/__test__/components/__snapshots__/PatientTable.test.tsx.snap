// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list vaccine page properly 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-start",
        "borderColor": "red",
        "borderWidth": 1,
      },
      undefined,
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#101010",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#e51c83",
            "borderRadius": 400,
            "fontSize": 40,
            "opacity": 0.7,
            "position": "absolute",
            "right": 40,
            "top": 1133.8999999999999,
            "width": "20%",
            "zIndex": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#fffffc",
              "fontSize": 40,
            }
          }
        >
          +
        </Text>
      </View>
      <View
        testID="card"
      >
        <RCTScrollView
          testID="card"
        >
          <View>
            <View
              style={
                Object {
                  "marginTop": 10,
                }
              }
            >
              <View
                testID="table"
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#171717",
                      "borderColor": "white",
                      "borderRadius": 50,
                      "borderWidth": 1.5,
                      "display": "flex",
                      "flexDirection": "row",
                      "flexWrap": "wrap",
                      "height": 60,
                      "margin": 13,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "uri": "asdfas",
                      }
                    }
                    style={
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "#bdbdbb",
                        "borderRadius": 500,
                        "display": "flex",
                        "height": 35,
                        "width": "10%",
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "left": 8,
                        "position": "relative",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fffffc",
                          "fontSize": 16,
                        }
                      }
                    >
                      first patient
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#bdbdbb",
                          "fontSize": 12,
                        }
                      }
                    >
                      asdfas
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      Object {
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignSelf": "flex-end",
                        "borderRadius": 500,
                        "marginLeft": "auto",
                        "opacity": 1,
                        "width": "20%",
                        "zIndex": 1,
                      }
                    }
                    testID="favouriteIcon"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Object {
                            "alignSelf": "center",
                            "color": "#e51c83",
                            "fontSize": 28,
                            "marginHorizontal": 10,
                          },
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰓒
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      Object {
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#e51c83",
                        "borderRadius": 500,
                        "marginLeft": 30,
                        "marginRight": 5,
                        "opacity": 1,
                        "width": "10%",
                        "zIndex": 10,
                      }
                    }
                    testID="deleteIcon"
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 18,
                          "textAlign": "center",
                        }
                      }
                    >
                      ╳
                    </Text>
                  </View>
                </View>
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#171717",
                      "borderColor": "white",
                      "borderRadius": 50,
                      "borderWidth": 1.5,
                      "display": "flex",
                      "flexDirection": "row",
                      "flexWrap": "wrap",
                      "height": 60,
                      "margin": 13,
                      "opacity": 1,
                      "padding": 5,
                    }
                  }
                >
                  <Image
                    source={
                      Object {
                        "uri": "asdfas",
                      }
                    }
                    style={
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "#bdbdbb",
                        "borderRadius": 500,
                        "display": "flex",
                        "height": 35,
                        "width": "10%",
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "left": 8,
                        "position": "relative",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#fffffc",
                          "fontSize": 16,
                        }
                      }
                    >
                      second patient
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#bdbdbb",
                          "fontSize": 12,
                        }
                      }
                    >
                      asdfas
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      Object {
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignSelf": "flex-end",
                        "borderRadius": 500,
                        "marginLeft": "auto",
                        "opacity": 1,
                        "width": "20%",
                        "zIndex": 1,
                      }
                    }
                    testID="favouriteIcon"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": undefined,
                            "fontSize": 12,
                          },
                          Object {
                            "alignSelf": "center",
                            "color": "#e51c83",
                            "fontSize": 28,
                            "marginHorizontal": 10,
                          },
                          Object {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      󰓒
                    </Text>
                  </View>
                  <View
                    accessibilityState={
                      Object {
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#e51c83",
                        "borderRadius": 500,
                        "marginLeft": 30,
                        "marginRight": 5,
                        "opacity": 1,
                        "width": "10%",
                        "zIndex": 10,
                      }
                    }
                    testID="deleteIcon"
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 18,
                          "textAlign": "center",
                        }
                      }
                    >
                      ╳
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
